addons/account/models/account_tax.py

```diff
@@ -862,6 +862,9 @@ def _compute_taxes_for_single_line(self, base_line, handle_price_include=True, i
 
             tax_values_list = []
             for tax_res in taxes_res['taxes']:
+                tax = self.env['account.tax'].browse(tax_res['id'])
+                if tax.amount_type == 'fixed' and early_pay_discount_computation:
+                    continue
```

Comment:
I'm not sure about this part. The filtering of fixed taxes could be in `_get_invoice_counterpart_amls_for_early_payment_discount_per_payment_term_line` before this method.

Reply:
Good idea thanks, doing this we reduce the diff by half